pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -589,11 +589,16 @@ func init() {
 					Verbs:     sets.NewString("list", "watch", "get"),
 					Resources: sets.NewString("storageclasses"),
 				},
-				// Gluster provisioner
+				// Glusterfs provisioner
 				{
 					Verbs:     sets.NewString("get", "create", "delete"),
 					Resources: sets.NewString("services", "endpoints"),
 				},
+				// Glusterfs & Ceph provisioner
+				{
+					Verbs:     sets.NewString("get"),
+					Resources: sets.NewString("secrets"),
```

Comment:
> only work off of storageclass objects, not PV annotations

Yes, StorageClass only. Kubernetes master (upcoming 1.5) does not use PV annotations and  all the necessary patches were ported to Origin 1.4. I don't think we used PV annotations in "vanilla" Kubernetes 1.4.